src/tool/hpcrun/ompt/ompt-interface.c

```diff
@@ -699,6 +698,41 @@ ompt_idle_blame_shift_request()
   ompt_register_idle_metrics();
 }
 
+//*****************************************************************************
+// map opid operations
+//*****************************************************************************
+
+// Records the (target id —> cct_node *) mapping in a thread local variable
+void
+hpcrun_opid_map_record_target(int target_id,
+                              cct_node_t *node)
+{
+  assert(target_id != 0);
+  ompt_region_map_insert((uint64_t) target_id, node);
+}
+
+
+// Adds an (opid, cct_node_t *) entry to the concurrent map by finding
+// the cct_node_t * associated with target_id from the thread_local variable
+void
+hpcrun_opid_map_insert(int host_op_id,
+                       int target_id)
+{
+  cct_node_t *node = ompt_region_map_entry_callpath_get(ompt_region_map_lookup(target_id));
+  ompt_host_op_map_insert(host_op_id, node);
```

Comment:
I copy the region map structure files and name it to host_op_map. We can do further optimizations on the host_op_map later.